File: Ruby/PE0006.rb
def square_of_sums(n=100)
  #(1..100).inject {|result, num| result + num}
  ((n*(n+1))/2)**2
end

def sum_of_squares(n=100)
  (1..n).inject{|result,num| result+(num*num)}
end


square = square_of_sums
sum = sum_of_squares
puts "square of sum: " + square.to_s
puts "sum of squares: " + sum.to_s
puts "difference: " +  (square-sum).to_s

File: Java/PE0006.java
public class PE0006 {

	public static void main(String[] args){

		int n = 100;

		int sum = sumOfSquares(n);
		int sq = squareOfSum(n);
		int diff = sq - sum;

		System.out.println("squareOfSum: " + sq);
		System.out.println("sumOfSquares: " + sum);
		System.out.println("Difference: " + diff);
	}


	public static int sumOfSquares(int n){
		int sum = 0;
		for(int i = 1; i <= n; i++){
			sum += i*i;
		}
		return sum;
	}

	public static int squareOfSum(int n){
		int sum = (n*(n+1))/2;
		return sum*sum;
	}
}
